src/inbox.rs

```diff
@@ -485,6 +466,35 @@ impl<A> ChanInner<A> {
     }
 }
 
+fn find_remove_highest_priority<M>(
```

Comment:
Related to: how I mentioned using BinaryHeap for priority messages, this could be removed & replaced with an O(log n) binary heap remove call